homeassistant/components/vizio/config_flow.py

```diff
@@ -127,12 +127,15 @@ def __init__(self) -> None:
             if entry.data[CONF_HOST] == import_config[CONF_HOST] and entry.data[
                 CONF_NAME
             ] == import_config.get(CONF_NAME):
-                if (
-                    CONF_VOLUME_STEP in import_config
-                    and entry.options[CONF_VOLUME_STEP]
-                    != import_config[CONF_VOLUME_STEP]
-                ):
-                    entry.options[CONF_VOLUME_STEP] = import_config[CONF_VOLUME_STEP]
+                if entry.data[CONF_VOLUME_STEP] != import_config[CONF_VOLUME_STEP]:
+                    new_volume_step = {
+                        CONF_VOLUME_STEP: import_config[CONF_VOLUME_STEP]
+                    }
+                    self.hass.config_entries.async_update_entry(
+                        entry=entry,
+                        data=entry.data.update(new_volume_step),
```

Comment:
Don't update the entry.data dict. Let the api call do that. We need to copy the data before we update it.

Reply:
thanks, that is what I was intending to do but moved too quickly